core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java

```diff
@@ -311,8 +311,8 @@ public boolean mkdirs(String path, MkdirsOptions options) throws IOException {
   }
 
   @Override
-  public InputStream open(String path, OpenOptions options) throws IOException {
-    return openObject(stripPrefixIfPresent(path), options);
+  public UnderFileInputStream open(String path, OpenOptions options) throws IOException {
+    return new ObjectUnderFileInputStream(this, stripPrefixIfPresent(path), options);
```

Comment:
Do you think you can just change ObjectUnderFileSystem to implements Seekable instead of having another class? 

If you really need another type, it is probably better to change the impl of this class to be deriving from that. 

Same comment for all other ufs-es